src/eva/data/jedi_log.py

```diff
@@ -0,0 +1,285 @@
+# (C) Copyright 2021-2023 NOAA/NWS/EMC
+#
+# (C) Copyright 2021-2023 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+import os
+import numpy as np
+import xarray as xr
+
+from eva.eva_base import EvaBase
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+# Parameters
+space = ' '
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+def get_data_from_line(jedi_log_line, search_term, separator, position):
+
+    if search_term in jedi_log_line:
+        return jedi_log_line.split(separator)[position]
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+class JediLog(EvaBase):
+
+    # ----------------------------------------------------------------------------------------------
+
+    def execute(self, data_collections, timing):
+
+        # Get name of the log file to parse
+        jedi_log_to_parse = self.config.get('jedi_log_to_parse')
+
+        # Collection name to use
+        collection_name = self.config.get('collection_name')
+
+        # Read log file into a string
+        with open(jedi_log_to_parse) as jedi_log_to_parse_open:
+            jedi_log_text = jedi_log_to_parse_open.read()
+
+        # Split log into list of lines
+        jedi_log_lines = jedi_log_text.split('\n')
+
+        # Check if this was a ctest and if so determine test prepend string
+        test_string = ''
+        for jedi_log_line in jedi_log_lines:
+            if jedi_log_line[0:4] == 'test':
```

Comment:
I thought it was `Test        :` like `CostJo    :` ? or am I thinking of something else?

Reply:
From what I can see when running ctest is prepends the log with a few lines. One of the first lines looks like e.g.:

`test 80`

We need to get `80` in order to know that all the other lines of the log will have `80: ` prepended to them. The next step is to remove that from all the lines so the subsequent parsing can proceed similarly to if the log came from a non ctest run, which would presumably be the standard.

Reply:
I believe the `Test        :` bit might be from when you output things with `oops::Log::test()`? As in [this line](https://github.com/JCSDA-internal/oops/blob/66c008ca38ae342fb8fda9b283cd16bb8c74c64c/src/oops/runs/ConvertState.h#L133). Then you get a line which looks like `80: Test     :`.

Reply:
ah okay, got it